client/app/core/rbac.service.js

```diff
@@ -1,32 +1,45 @@
+import RBACActions from '../actions/rbac.actions'
+import NavActions from '../actions/nav.actions'
 /** @ngInject */
-export function RBACFactory (lodash) {
+export function RBACFactory (lodash, $ngRedux) {
   var navFeatures = {}
   let features = {}
   let currentRole
 
-  return {
+  const service = {
     all: all,
     set: set,
     has: has,
     hasAny: hasAny,
     hasRole: hasRole,
     setRole: setRole,
     getNavFeatures: getNavFeatures,
-    setNavFeatures: setNavFeatures,
     navigationEnabled: navigationEnabled
   }
 
+  function mapStateToThis (state) {
+    return {
+      rbac: state.rbac,
+      nav: state.nav
+    }
+  }
+
```

Comment:
nitpick: AFAIR (not tested) you can write this as
```js
function mapStateToThis({ rbac, nav }) {
  return { rbac,  nav };
}
```

Reply:
@ohadlevy , I will definitely take a look at all your feedback.  I am actually planning on redoing the work in this branch.  After I got started I came to realize that a whole bunch of tech debt needed to be settled up before we proceeded much further with redux.  A lot of this stuff was prototype code and definitely not up to snuff.  I am working on this reimplementation as we speak.  Thank you for taking a look.  I am sure before this goes that much further, I will need some of your expertise in ensuring we implement based on best practices you have experienced.